runtime/src/main/java/org/corfudb/runtime/collections/CorfuStore.java

```diff
@@ -93,23 +93,53 @@ private Timestamp getTimestamp() {
     }
 
     /**
-     * Creates and registers a table.
-     * A table needs to be registered before it is used.
-     *
-     * @param namespace    Namespace of the table.
-     * @param tableName    Table name.
-     * @param kClass       Class of the Key Model.
-     * @param vClass       Class of the Value Model.
-     * @param mClass       Class of the Metadata Model.
-     * @param tableOptions Table options.
-     * @param <K>          Key type.
-     * @param <V>          Value type.
-     * @param <M>          Type of Metadata.
-     * @return Table instance.
-     * @throws NoSuchMethodException     Thrown if key/value class are not protobuf classes.
-     * @throws InvocationTargetException Thrown if key/value class are not protobuf classes.
-     * @throws IllegalAccessException    Thrown if key/value class are not protobuf classes.
+     * Subscribe to the routing queue notifications arriving from a remote cluster.
      */
+    public void subscribeRoutingQListener(@Nonnull LiteRoutingQueueListener routingQueueListener) {
+        Timestamp ts = routingQueueListener.performFullSync();
+        this.subscribeListener(routingQueueListener, CORFU_SYSTEM_NAMESPACE, REPLICATED_QUEUE_TAG,
+                Arrays.asList(LogReplicationUtils.REPLICATED_RECV_Q_PREFIX + routingQueueListener.getSourceSiteId() +
+                    "_" + routingQueueListener.getClientName()), ts);
+    }
+
+    public static <K extends Message, V extends Message, M extends Message>
+    ScopedTransaction snapshotFederatedTables(String namespace, CorfuRuntime runtime) {
+        ArrayList<Table> federatedTables = new ArrayList<>();
+        runtime.getTableRegistry().getAllOpenTables().forEach(t -> {
+            String tableNameWithoutNs = StringUtils.substringAfter(t.getFullyQualifiedTableName(),
+                    t.getNamespace() + "$");
+            CorfuStoreMetadata.TableName tableName = CorfuStoreMetadata.TableName.newBuilder()
+                    .setNamespace(t.getNamespace())
+                    .setTableName(tableNameWithoutNs).build();
+            CorfuRecord<CorfuStoreMetadata.TableDescriptors, CorfuStoreMetadata.TableMetadata> tableRecord =
+                    runtime.getTableRegistry().getRegistryTable().get(tableName);
+            if (tableRecord.getMetadata().getTableOptions().getIsFederated()) {
+                federatedTables.add(t);
+            }
+        });
+        return new ScopedTransaction(runtime, namespace,
```

Comment:
Hi @vjeko - sorry if I missed it but I don't see the retry logic for the possible transient failure that can come from the corfu's implementation of MVCC? Can you please share where the scope transaction creation is retried?

Reply:
Looks like I missed that path. Fixed.